lib/rules/lint-no-childless-elements.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+/*
+ Disallows self-closing void elements
+
+ ```
+ {{! good }}
+ <hr>
+
+ {{! bad}}
+ <hr />
+ ```
+
+ The following values are valid configuration:
+
+   * boolean -- `true` for enabled / `false` for disabled
+ */
+
+const Rule = require('./base');
+const createErrorMessage = require('../helpers/create-error-message');
+
+/**
+ * [Specs of Void Elements]{@link https://www.w3.org/TR/html-markup/syntax.html#void-element}
```

Comment:
This should go in the `References` section of the doc.

Here's the doc template: https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/_TEMPLATE_.md